atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/TimestampChecker.java

```diff
@@ -41,27 +52,32 @@ private TimestampChecker() {
      *     :errors     A list of events that failed the check, or an empty list if the check passed
      * @throws Exception if the parsing of the history fails.
      */
-    public static Map<Keyword, Object> checkClojureHistory(List<Map<Keyword, ?>> clojureHistory) {
+    public Map<Keyword, Object> checkClojureHistory(List<Map<Keyword, ?>> clojureHistory) {
         List<Event> events = convertClojureHistoryToEventList(clojureHistory);
         return checkHistory(events);
     }
 
-    private static List<Event> convertClojureHistoryToEventList(List<Map<Keyword, ?>> clojureHistory) {
+    private List<Event> convertClojureHistoryToEventList(List<Map<Keyword, ?>> clojureHistory) {
         return clojureHistory.stream()
                 .map(Event::fromKeywordMap)
                 .collect(Collectors.toList());
     }
 
-    private static Map<Keyword, Object> checkHistory(List<Event> events) {
-        MonotonicChecker monotonicChecker = new MonotonicChecker();
-        events.forEach(event -> event.accept(monotonicChecker));
-        return createMapFromCompletedChecker(monotonicChecker);
+    private Map<Keyword, Object> checkHistory(List<Event> events) {
```

Comment:
I can't have a function body in an interface; so I'd have to start subclassing? Something like the below? TBH I don't think it's worth it...
```
interface EventVisitor
Class Checker implements EventVisitor {
void check(List<Event> events) {
 events.foreach...
}
class MonotonicChecker extends Checker {
}
```